src/SFML/Graphics/Font.cpp

```diff
@@ -639,11 +656,38 @@ Glyph Font::loadGlyph(std::uint32_t codePoint, unsigned int characterSize, bool
         width += 2 * padding;
         height += 2 * padding;
 
-        // Get the glyphs page corresponding to the character size
-        Page& page = loadPage(characterSize);
+        // Get the pages list corresponding to the character size
+        PageList& pages = m_pageLists[characterSize];
 
-        // Find a good position for the new glyph into the texture
-        glyph.textureRect = findGlyphRect(page, {width, height});
+        // Find a page that can fit well the glyph
+        Page* page = NULL;
+        for (PageList::iterator it = pages.begin(); it != pages.end() && !page; ++it)
+        {
+            // Try to find a good position for the new glyph into the texture
+            glyph.textureRect = findGlyphRect(*it, {width, height});
+
+            if (glyph.textureRect.width > 0)
+                page = &*it;
```

Comment:
This assigns to `glyph.textureRect` even if no good rect is found, and leaves this assigned beyond the `for` loop. I see that you have another `glyph.textureRect = findGlyphRect(*page, {width, height});` further down, so it's correct.

However, this flow is very prone to errors on refactorings. The function is also getting rather big, and variable mutations are mixed with "found" state. A separate function could make things easier:
```cpp
struct MatchingPage {
    Page* page = nullptr;
    IntRect textureRect;
};

std::optional<MatchingPage> findMatchingPage(...);
```
```cpp
// usage:
if (auto matching = findMatchingPage(...)) {
    page = matching->page;
    glyph.textureRect = matching->textureRect;
    ...
} else {
    ...
}
```

Reply:
Personally I think this would over-engineer it for little gain - the behaviour is only required in one place so I would prefer to use a lambda limited to this scope, but then I don't think that would really address any of your concerns.

Would like to hear other opinions on this though